### PR TITLE
Container image vulnerability scanning

### PR DESCRIPTION
**Pull request by aliaksei-ivanou**
*Issue #, if available:*

*Description of changes:*

Creating the pattern demonstrating how to enable and configure Amazon ECR image scanning.

The following scanning types are offered.

- Enhanced scanning—Amazon ECR integrates with Amazon Inspector to provide automated, continuous scanning of your repositories. Your container images are scanned for both operating systems and programing language package vulnerabilities. As new vulnerabilities appear, the scan results are updated and Amazon Inspector emits an event to EventBridge to notify you.
- Basic scanning—Amazon ECR uses the Common Vulnerabilities and Exposures (CVEs) database from the open-source Clair project. With basic scanning, you configure your repositories to scan on push or you can perform manual scans and Amazon ECR provides a list of scan findings.

The pattern consists of two components:

- `ImageScanningSetupStack` - configures the Amazon ECR image scanning and the ECR automated re-scan duration in Inspector.
- A blueprint that deploys a sample GitOps workload that pushes images to Amazon ECR and triggers the image scanning.

Please see the documentation in the PR for more information.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/patterns/security/image-scanning.md [hunk 1]
```diff
@@ -0,0 +1,388 @@
+# Amazon ECR Image Scanning
+
+## Objective
+
+The objective of this pattern is to demonstrate how to enable and configure Amazon ECR image scanning.
+
+The following scanning types are offered.
```
Comment (shapirov103) [inline docs/patterns/security/image-scanning.md:7]: minor: `:` at the end of this sentence.

File: docs/patterns/security/image-scanning.md [hunk 2]
```diff
@@ -0,0 +1,388 @@
...
+- A blueprint that deploys a sample GitOps workload that pushes images to Amazon ECR and triggers the image scanning.
+
+## Configuration
+
+You can configure the following parameters in the [ImageScanningSetupStack](../../../lib/security/image-vulnerability-scanning/image-scanning-setup.ts) stack:
+
+- `scanType` - The type of scan to perform. Valid values are `BASIC` and `ENHANCED`.
+- Enhanced scanning only:
+  - `enhancedContinuousScanDuration` - the Amazon ECR automated re-scan duration setting determines how long Amazon Inspector continuously monitors images pushed into repositories. When the number of days from when an image is first pushed exceeds the automated re-scan duration configuration, Amazon Inspector stops monitoring the image. When Amazon Inspector stops monitoring an image, the scan status of the image is changed to inactive with a reason code of expired, and all associated findings for the image are scheduled to be closed. Valid values are `LIFETIME`, `DAYS_30`, and `DAYS_180`.
+  - `enchancedScanRules` - the scanning rules.
```
Comment (shapirov103) [inline docs/patterns/security/image-scanning.md:24]: minor: enchancedScanRules - assume you meant enhancedScanRules

File: docs/patterns/security/image-scanning.md [hunk 3]
```diff
@@ -0,0 +1,388 @@
...
+
+For GitOps, the blueprint bootstraps the ArgoCD addon and points to the [EKS Blueprints Workload](https://github.com/aws-samples/eks-blueprints-workloads) sample repository.
+
+The sample repository contains the following workloads:
+
+- `team-scan` pushes a Docker image to Amazon ECR and triggers the image scanning.
+
+## Prerequisites
+
+1. Clone the repository
```
Comment (shapirov103) [inline docs/patterns/security/image-scanning.md:40]: Numbered list, expected it to be 1.2.3 unless you tell me MD renders correctly regardless.

File: docs/patterns/security/image-scanning.md [hunk 4]
```diff
@@ -0,0 +1,388 @@
...
+
+```bash
+npx cdk deploy image-scanning-setup
+```
+
+### Deploying the blueprint
+
+The blueprint deploys a sample GitOps workload that pushes images to Amazon ECR and triggers the image scanning.
+
+To deploy the bluepring, run the following command:
```
Comment (shapirov103) [inline docs/patterns/security/image-scanning.md:60]: minor: bluepring

File: docs/patterns/security/image-scanning.md [hunk 5]
```diff
@@ -0,0 +1,388 @@
...
+
+```bash
+npx cdk deploy image-scanning-workload-blueprint
+```
+
+## Verify
+
+### Verifying that the image scanning is enabled
+
+To verify that the image scanning is enabled on the registry level, run the following command:
```
Comment (shapirov103) [inline docs/patterns/security/image-scanning.md:70]: minor: "on the registry level" => "at the registry level" or just "on the registry" without level.

File: docs/patterns/security/image-scanning.md [hunk 6]
```diff
@@ -0,0 +1,388 @@
...
+    ]
+}
+```
+
+### Checking the image scanning findings
+
+To check the image scanning findings, run the following command:
+
+```bash
+aws ecr describe-image-scan-findings --repository-name image-scanning-workload-blueprint-imagescanningrepository754c6116-arh0wk3afnkw --image-id imageTag=latest
```
Comment (shapirov103) [inline docs/patterns/security/image-scanning.md:142]: call out that the repository name will have to be replaced with their specific name.

File: docs/patterns/security/image-scanning.md [hunk 7]
```diff
@@ -0,0 +1,388 @@
...
+                    }
+                },
+                "severity": "MEDIUM",
+                "status": "ACTIVE",
+                "title": "CVE-2021-33910 - systemd-pam, systemd and 2 more",
+                "type": "PACKAGE_VULNERABILITY",
+                "updatedAt": "2023-04-17T17:40:39.940000-05:00"
+            },
+        }
+...
```
Comment (shapirov103) [inline docs/patterns/security/image-scanning.md:267]: trim down the findings (or if there is a collapsible box, let's allocated 10 lines max). What is this stuff in red here? Does not render correctly when you preview the markdown (e.g open preview in vscode)?

File: docs/patterns/security/image-scanning.md [hunk 8]
```diff
@@ -0,0 +1,388 @@
...
+You can also check the findings in Inspector2.
+
+```bash
+aws inspector2 list-findings
+```
+
+The output should look similar to the following:
+
+```json
+{
```
Comment (shapirov103) [inline docs/patterns/security/image-scanning.md:279]: minor: same, trim down the findings and/or use a collapsible box if there is one with 10 lines max.

File: lib/security/image-vulnerability-scanning/index.ts [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+    await prevalidateSecrets(ImageScanningWorkloadConstruct.name, process.env.CDK_DEFAULT_REGION!, SECRET_ARGO_ADMIN_PWD);
+
+    const ecrRepositoryName = "ImageScanningRepository";
+    const ecrRepository = blueprints.getNamedResource(
+      ecrRepositoryName
+    ) as ecr.Repository;
+
+    const stackID = `${id}-blueprint`;
+    await blueprints.EksBlueprint.builder()
+      .account(process.env.CDK_ACCOUNT_ID!)
```
Comment (shapirov103) [inline lib/security/image-vulnerability-scanning/index.ts:23]: where is that env variable from `CDK_ACCOUNT_ID`? we use `CDK_DEFAULT_ACCOUNT` across the board.

**Review by shapirov103 — COMMENTED** [review]
Some minor comments, please address, I will test out.

File: docs/patterns/security/image-scanning.md (hunk 3 above)
Comment (aliaksei-ivanou) [reply docs/patterns/security/image-scanning.md:40]: Yes, I believe so https://www.markdownguide.org/basic-syntax/#ordered-lists

File: docs/patterns/security/image-scanning.md (hunk 1 above)
Comment (aliaksei-ivanou) [reply docs/patterns/security/image-scanning.md]: Addressed

File: docs/patterns/security/image-scanning.md (hunk 2 above)
Comment (aliaksei-ivanou) [reply docs/patterns/security/image-scanning.md]: Addressed

File: docs/patterns/security/image-scanning.md (hunk 4 above)
Comment (aliaksei-ivanou) [reply docs/patterns/security/image-scanning.md]: Addressed

File: docs/patterns/security/image-scanning.md (hunk 5 above)
Comment (aliaksei-ivanou) [reply docs/patterns/security/image-scanning.md]: Addressed

File: docs/patterns/security/image-scanning.md (hunk 6 above)
Comment (aliaksei-ivanou) [reply docs/patterns/security/image-scanning.md]: Addressed

**Review by shapirov103 — APPROVED** [review]
LGTM

**Review by shapirov103 — APPROVED** [review]
Validated both stacks, observed inspector findings for the offending image.

**Pull request merged**